### PR TITLE
feat(unstable): "Deno.Command()" API

### PR DESCRIPTION
**Pull request by crowlKats**
Easy async
```ts
let c = new Deno.Command("echo", { args: ["foo"] });
let { stdout, stderr } = await c.output();
```

Easy sync
```ts
let c = new Deno.Command("echo", { args: ["foo"] });
let { stdout, stderr } = c.outputSync();
```

Everything else
```ts
const c = new Deno.Command("cat", { stdin: "piped" });
c.spawn();
c.stdout.pipeTo(Deno.openSync("output", { write: true }).writable);

const x = c.stdin.getWriter();
await x.write(new TextEncoder().encode("foobar"));
await x.close();

const s = await c.status;
console.log(s);
```

### CODE REVIEW

**Comment by load1n9** [comment]
I approve

**Comment by lucacasonato** [comment]
`Deno.command` should instead be `new Deno.Command()`. JS doesn’t really do synchronous factory functions anymore

**Comment by crowlKats** [comment]
@lucacasonato that was originally the plan for what now is the spawn API, however there was pushback on that

**Comment by lucacasonato** [comment]
Is this meant to be a replacement for Deno.spawn?

**Comment by ry** [comment]
Needs tests

**Comment by crowlKats** [comment]
@ry there already are tests

**Review by ry — APPROVED** [review]
LGTM

File: cli/diagnostics.rs [hunk 1]
```diff
@@ -33,8 +33,14 @@ const UNSTABLE_DENO_PROPS: &[&str] = &[
   "Child",
   "spawn",
   "spawnSync",
+  "SpawnOptions",
   "ChildStatus",
   "SpawnOutput",
+  "command",
```
Comment (lucacasonato) [inline cli/diagnostics.rs:39]: Not needed

**Review by dsherret — APPROVED** [review]
LGTM!

**Comment by sigmaSd** [comment]
I don't understand, I like the spawn API, what does this do better?

**Comment by dsherret** [comment]
@sigmaSd the spawn api was three separate functions (`spawn`, `spawnSync`, and `spawnChild`) and the meaning of `spawn` and `spawnChild` was ambiguous. This reduces the ambiguity and consolidates everything under a single API.

**Comment by sigmaSd** [comment]
I see thanks for the info

**Pull request merged**